### DISCUSSION TITLE
Car object with its own policy

### DISCUSSION THREAD

**Discussion by lev-smolentsev-K1** [metadriverse/metadrive#712] [Q&A]
Hi, I wanted to try to create a car agent with its LineChange policy to change lines, during scenario
but i did not get how to span a distinct agent with its distinct policy.
Could you provide a minimal example where some agents are controlled by some "global" env policy and one or two are guided by other policy, like 
```
v = env.engine.spawn_object(DefaultVehicle, 
                                    vehicle_config=cfg, 
                                    position=[30,0], 
                                    heading=0,)
        env.agent_manager.add_policy(v.name,LaneChangePolicy,v.name,1)
```
and during simulation to do similar to:

```
p = env.engine.get_policy(v.name)
v.before_step(p.act(v.name))
_ , info = env.step([-1,-1])
```

**Comment by lev-smolentsev-K1** [comment]
In this case by agent I mean traffic vehicles. LaneChangePolicy only worked for an ego agent while in traffic_manager it throws: **TypeError: act() missing 1 required positional argument: 'agent_id'** for the following lines of code:

 ```
 for v in self._traffic_vehicles:
            p = self.engine.get_policy(v.name)
            v.before_step(p.act())
```

**Comment by lev-smolentsev-K1** [comment]
Could you provide any suggestions on how to customize the behavior of the traffic in your simulator?

**Comment by QuanyiLi** [comment]
For this aim, you need to customize the trafficManager and replace the original one with it. Just check more about managers in our documentation
